nucliadb/src/nucliadb/search/search/chat/query.py

```diff
@@ -181,11 +182,17 @@ async def run_main_query(
     find_request.debug = item.debug
     find_request.rephrase = item.rephrase
     find_request.rephrase_prompt = parse_rephrase_prompt(item)
+    # predict reranker will be done in their stream to avoid a round trip
+    # between nucliadb and predict
+    if item.reranker != Reranker.PREDICT_RERANKER:
+        find_request.reranker = item.reranker
     # We don't support pagination, we always get the top_k results.
-    find_request.page_size = item.top_k
-    find_request.page_number = 0
+    find_request.top_k = item.top_k
     find_request.show_hidden = item.show_hidden
 
+    # this executes the model validators, that can tweak some fields
+    FindRequest.model_validate(find_request)
```

Comment:
mh, interesting. can you elaborate a bit more?

shouln't it be:
`find_request = FindRequest.model_validate(find_request)`
?

Reply:
No, the validation modifies the object in-place.

```python
>>> from nucliadb_models.search import FindRequest
>>> req = FindRequest()
>>> req
FindRequest(..., page_size=20, top_k=None, ...)
>>> FindRequest.model_validate(req)
FindRequest(..., page_size=10, top_k=10, ...)
>>> req
FindRequest(..., page_size=10, top_k=10, ...)
```